collect_app/src/main/java/org/odk/collect/android/projects/AutomaticProjectCreatorDialog.kt

```diff
@@ -0,0 +1,163 @@
+import org.odk.collect.android.views.BarcodeViewDecoder
+import org.odk.collect.material.MaterialFullScreenDialogFragment
+import org.odk.collect.projects.R
+import javax.inject.Inject
+
+class AutomaticProjectCreatorDialog : MaterialFullScreenDialogFragment() {
+
+    @Inject
+    lateinit var barcodeViewDecoder: BarcodeViewDecoder
+
+    @Inject
+    lateinit var permissionsProvider: PermissionsProvider
+
+    @Inject
+    lateinit var projectCreator: ProjectCreator
+
+    private var capture: CaptureManager? = null
+
+    private lateinit var beepManager: BeepManager
+    lateinit var binding: AutomaticProjectCreatorDialogLayoutBinding
+
+    private var listener: ProjectAddedListener? = null
+
+    override fun onAttach(context: Context) {
+        super.onAttach(context)
+        DaggerUtils.getComponent(context).inject(this)
+
+        if (context is ProjectAddedListener) {
+            listener = context
+        }
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
+        binding = AutomaticProjectCreatorDialogLayoutBinding.inflate(inflater)
+        setUpToolbar()
+        binding.configureManuallyButton.setOnClickListener {
+            DialogUtils.showIfNotShowing(ManualProjectCreatorDialog::class.java, requireActivity().supportFragmentManager)
+            lifecycleScope.launch {
+                delay(1000)
+                dismiss()
+            }
+        }
+        binding.cancelButton.setOnClickListener {
+            dismiss()
+        }
+        beepManager = BeepManager(requireActivity())
+        permissionsProvider.requestCameraPermission(
+            requireActivity(),
+            object : PermissionListener {
+                override fun granted() {
+                    startScanning(savedInstanceState)
+                }
+
+                override fun denied() {
+                    dismiss()
+                }
+            }
+        )
+        return binding.root
+    }
+
+    override fun onCloseClicked() {
+    }
+
+    override fun onBackPressed() {
+        dismiss()
+    }
+
+    override fun onSaveInstanceState(outState: Bundle) {
+        capture?.onSaveInstanceState(outState)
+        super.onSaveInstanceState(outState)
+    }
+
+    override fun onPause() {
+        super.onPause()
+        binding.barcodeView.pauseAndWait()
+        capture?.onPause()
+    }
+
+    override fun onResume() {
+        super.onResume()
+        binding.barcodeView.resume()
+        capture?.onResume()
+    }
+
+    override fun onDestroy() {
+        capture?.onDestroy()
+        super.onDestroy()
+    }
+
+    override fun getToolbar(): Toolbar {
+        return binding.toolbar
+    }
+
+    private fun setUpToolbar() {
+        toolbar.setTitle(R.string.add_project)
+        toolbar.navigationIcon = null
+    }
+
+    private fun startScanning(savedInstanceState: Bundle?) {
+        val intent = IntentIntegrator(requireActivity())
```

Comment:
Are we able to wrap all the shared (between this and `BarcodeScannerFragment`) setup for the barcode scanning? Even if it's not all tested, it would be great to have it isolated, so it's easy to debug inspect in one place.

Reply:
Done: https://github.com/getodk/collect/pull/4591/commits/a3a164b142efe1e10a2119a9fdf2e58c89a6c367